### PR TITLE
[Connect] Add ability to specify arbitrary demo accounts in example app

### PR DESCRIPTION
**Pull request by mludowise-stripe**
## Summary

Adds the ability specify arbitrary demo / test accounts in the Connect example app.

Primary use case is so we can test with different accounts internally during bug-bashing, but also useful for testing onboarding since we need to create new accounts with empty data to test it properly.

## Motivation

https://jira.corp.stripe.com/browse/MXMOBILE-2793

## Testing

https://github.com/user-attachments/assets/6ad28c68-dd1b-4025-90d2-798cf2d31ec4

### CODE REVIEW

File: Example/StripeConnectExample/StripeConnectExample/AppSettingsView.swift [hunk 1]
```diff
@@ -8,32 +8,56 @@
...
+        }
+        return !appInfo.availableMerchants.contains(selectedMerchant)
+    }
+
+    var isMerchantIdValid: Bool {
+        if !isUsingCustomMerchant {
+            return true
+        } else if let selectedMerchant {
+            return selectedMerchant.id.starts(with: "acct_")
+            && selectedMerchant.id.count > 5
```
Comment (mats-stripe) [inline Example/StripeConnectExample/StripeConnectExample/AppSettingsView.swift:39]:
```suggestion
            return isMerchantIdValid(selectedMerchant.id)
```

**Review by mats-stripe — APPROVED** [review]
Drive-by review!

File: Example/StripeConnectExample/StripeConnectExample/AppSettingsView.swift (hunk 1 above)
Comment (mludowise-stripe) [reply Example/StripeConnectExample/StripeConnectExample/AppSettingsView.swift:39]: Oops, I meant to delete `isMerchantIdValid(_ id)` 🤦‍♀️

**Review by nschris-stripe — DISMISSED** [review]

**Review by nschris-stripe — APPROVED** [review]
